examples/guess-the-sketch/matchmaker/director/director.go

```diff
@@ -0,0 +1,193 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/anypb"
+	"open-match.dev/open-match/pkg/pb"
+
+	allocationv1 "agones.dev/agones/pkg/apis/allocation/v1"
+	"agones.dev/agones/pkg/client/clientset/versioned"
+	"google.golang.org/grpc/credentials/insecure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	omApiHost  = "open-match-backend.open-match.svc.cluster.local:50505"
+	mmfApiHost = "guess-the-sketch-mmf.genai.svc.cluster.local"
+	mmfApiPort = 50502
+)
+
+type Client struct {
+	BackendServiceClient       pb.BackendServiceClient
+	CloserBackendServiceClient func() error
+	AgonesClientset            versioned.Interface
+}
+
+func main() {
+	log.Println("Starting Director")
+
+	for range time.Tick(time.Second) {
+
+		omBackendClient, omCloser := createOMBackendClient()
+
+		var r Client
+		r.AgonesClientset = createAgonesClient()
+		r.BackendServiceClient = omBackendClient
+		r.CloserBackendServiceClient = omCloser
+
+		if err := r.run(); err != nil {
+			log.Println("Error running director:", err.Error())
+		}
+	}
+}
+
+func createOMBackendClient() (pb.BackendServiceClient, func() error) {
+	conn, err := grpc.Dial(omApiHost, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		panic(err)
+	}
+	return pb.NewBackendServiceClient(conn), conn.Close
+}
+
+func createAgonesClient() *versioned.Clientset {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+	agonesClient, err := versioned.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	return agonesClient
+}
+
+// Customize the backend.FetchMatches request, the default one will return all tickets in the statestore
+func createOMFetchMatchesRequest() *pb.FetchMatchesRequest {
+	return &pb.FetchMatchesRequest{
+		// om-function:50502 -> the internal hostname & port number of the MMF service in our Kubernetes cluster
+		Config: &pb.FunctionConfig{
+			Host: mmfApiHost,
+			Port: mmfApiPort,
+			Type: pb.FunctionConfig_GRPC,
+		},
+		Profile: &pb.MatchProfile{
+			Name: "1v1",
+			Pools: []*pb.Pool{
+				{
+					Name: "everyone",
+				},
+			},
+			Extensions: map[string]*anypb.Any{},
+		},
+	}
+}
+
+func createAgonesGameServerAllocation() *allocationv1.GameServerAllocation {
+	// TODO: Update allocation requirements
```

Comment:
you probably don't need any

Reply:
Sounds good. I talked to Max and he mentioned that the only requirement he has is to make sure it allocates a game server in the Ready state only, which is the default.